ArtifactHandling/Resolve-DependenciesFromAzureFeed.ps1

```diff
@@ -0,0 +1,117 @@
+<# 
+ .Synopsis
+  Function for resolving dependencies using an AzureFeed
+ .Description
+  Function for resolving dependencies using an AzureFeed
+  To being able to resolve dependencies an artifact must be in the following format.
+  The app id must be in the artifact name so apps will be identified correctly.
+  The artifact can contain an .runtime.app file. If no app.json or .app file is present the .runtime.app fill be treated as a leaf in the tree.
+  If an app.json is present dependencies will be take from that file.
+  If no app.json but an .app file is present that will be extracted and used to find dependencies. 
+#>
+function Resolve-DependenciesFromAzureFeed {
+    Param(
+        [Parameter(Mandatory = $true)]
+        [string] $organization,
+        [Parameter(Mandatory = $true)]
+        [string] $feed,
+        [Parameter(Mandatory = $true)]
+        [string] $appsPath,
+        [string] $outPath = (Join-Path $appsPath '.alpackages'),
+        [string] $spaces = ''
+    )
+
+    $spaces += '   '
+
+    # Create outPath if not exits 
+    if (!(Test-Path $outPath)) {
+        New-Item -Path $outPath -ItemType Directory -Force | Out-Null
+    }
+
+    # Search for app.jsons
+    $apps = @(Get-ChildItem -Path (Join-Path $appsPath '/*/app.json'))
+    if ($apps.Count -eq 0) {
+        # Look for a single app.json in root
+        try {
+            $apps = @(Get-ChildItem -Path (Join-Path $appsPath '/app.json'))
+        } catch {}
+    }
+    if ($apps.Count -eq 0) {
+        # If no app.jsons found look for .app files.
+        $apps = @(Get-ChildItem -Path (Join-Path $appsPath '*.app') -Exclude '*.runtime.app')
+    }
+
+    Write-Host "$($spaces)$($apps.Count) apps found";
+
+    $apps | % {
+
+        if($_.Name -eq "app.json") {
+            Write-Host "$($spaces)$($_.DirectoryName | Split-Path -Leaf)"
+        } else {
+            Write-Host "$($spaces)$($_.Name)"
+        }
+
+        # Create temp folders
+        $tempAppFolder = Join-Path ((Get-Item -Path $env:temp).FullName) ([Guid]::NewGuid().ToString())
+        $tempAppDependenciesFolder = Join-Path $tempAppFolder 'dependencies'
+
+        New-Item -path $tempAppFolder -ItemType Directory -Force | Out-Null
```

Comment:
You should add a try/finally when creating a temp folder - to clean up is something fails